HdrHistogram/Iteration/AbstractHistogramEnumerator.cs

```diff
@@ -34,8 +34,13 @@ internal abstract class AbstractHistogramEnumerator : IEnumerator<HistogramItera
         protected int CurrentSubBucketIndex { get; private set; }
         protected long TotalCountToCurrentIndex { get; private set; }
         protected long CountAtThisValue { get; private set; }
-
-        public HistogramIterationValue Current { get; private set; }
+        
+        private HistogramIterationValue _current;
+        public HistogramIterationValue Current 
+        { 
+            get         => _current.Clone(); 
```

Comment:
I can see the confusion here, but this is a low allocation path. Now that you are cloning (instead of using the value inplace) you are now allocating.
Consumers are welcome to execute a memberwise clone as they fetch the next value.

Reply:
Actually(!), having just run the benchmarks on `master` and this PR, there is no significant performacne change (improvement or degradation), so the spirit of this PR is valid.
:-)